source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,507 @@
+    Moving around in a file
+    Search / replace
+    Text manipulation, ie: cw, dw, c$, yy / p, x, .
+
+.. figure:: ../_static/vim_cheatsheet.gif
+    :scale: 75%
+
+Moving Around
+-------------
+
+::
+
+    h move one character to the left.
+    j move down one line.
+    k move up one line.
+    l move one character to the right.
+    0 move to the beginning of the line.
+    $ move to the end of the line.
+    w move forward one word.
+    b move backward one word.
+    G move to the end of the file.
+    gg move to the beginning of the file.
+    . move to the last edit.
+
+Configuration / customization
+-----------------------------
+
+.. note:: there are many many options and pre-existing packages to make
+    editing nice for sysadmins and developers
+
+* ``.vimrc``
+* ``:set``
+
+Some sets of Vim plugins and configurations are available
+
+* https://github.com/astrails/dotvim
+* https://github.com/carlhuda/janus
+
+Use them for research on what's available to improve dev productivity
+
+Learning Resources
+------------------
+
+* ``$ vimtutor``
+* http://vim-adventures.com/
+
+.. figure:: ../_static/learning_curves.jpg
+    :align: center
+    :scale: 140%
+
+Regular expressions
+-------------------
+
+You should know basic substitution:
+
+::
+
+    :%s/foo/bar/g
+
+On IRC, Hamper does rudimentary regex in the form ``s/foo/bar/`` applying only
+to the most recent comment.
+
+This is **not** `shell globbing`_
+
+:Resources for learning:
+  * `RegExr`_ - an interactive Regular Expression editor and debugger
+  * `Regular-Expressions.info`_ - Tutorials and general information
+
+.. _shell globbing: http://tldp.org/LDP/abs/html/globbingref.html
+.. _RegExr: http://gskinner.com/RegExr/
+.. _Regular-Expressions.info: http://www.regular-expressions.info/
+
+
+
+Emacs Moving Around
+-------------------
+
+::
+
+    C-f            forward one char (right)
+    C-b            backwards one char (left)
+    M-f            forward one word
+    M-b            backwards one word
+    C-n            forward one line (down)
+    C-p            backwards one line (up)
+    C-a            beginning of line
+    C-e            end of line
+    C-o            insert-newline and stay on current line
+    C-j            insert newline and indent
+    C-v            page down
+    M-v            page up
+    M-<            beginning of file
+    M->            end of file
+    M-g g <number> goto line <number>
+    C-s            forward search (C-s to keep searching)
+
+Emacs Buffers
+-------------
+
+* Like a tab on a browser
+* Each file gets a buffer
+* Special buffers begin and end with ``*``
+
+|
+
+::
+
+    C-x b switch buffers (type a new name to open a new buffer)
+    C-x C-b list all buffers
+    C-x C-f find file (opens a new buffer for the file)
+    C-x k kill buffer
+    C-x 1 close all windows but the main one
+    C-x 2 split window horizontally
+    C-x 3 split window vertically
+    C-x o switch window
+
+Emacs Modes
+-----------
+
+|
```

Comment:
Any particular reason why you keep adding an extra space on the following slides?